From 227157bf7386d17b8938ac961bd4e239047be194 Mon Sep 17 00:00:00 2001
From: Guoqing Jiang <guoqing.jiang@suse.com>
Date: Thu, 1 Sep 2022 10:29:52 +0800
Subject: [PATCH] Revert "xfs: only bother with sync_filesystem during readonly
 remount"
Patch-mainline: Never, revert stable patch which is added again in sorted section
References: bsn#83
Modified-by-SEL: No


This reverts commit 70d560e2fb5e4505121f58ceb0df5b1f905c71cc.

Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/xfs/xfs_super.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/fs/xfs/xfs_super.c b/fs/xfs/xfs_super.c
index 434c87cc9fbf..ff686cb16c7b 100644
--- a/fs/xfs/xfs_super.c
+++ b/fs/xfs/xfs_super.c
@@ -1720,11 +1720,6 @@ xfs_remount_ro(
 	};
 	int			error;
 
-	/* Flush all the dirty data to disk. */
-	error = sync_filesystem(mp->m_super);
-	if (error)
-		return error;
-
 	/*
 	 * Cancel background eofb scanning so it cannot race with the final
 	 * log force+buftarg wait and deadlock the remount.
@@ -1795,6 +1790,8 @@ xfs_fc_reconfigure(
 	if (error)
 		return error;
 
+	sync_filesystem(mp->m_super);
+
 	/* inode32 -> inode64 */
 	if ((mp->m_flags & XFS_MOUNT_SMALL_INUMS) &&
 	    !(new_mp->m_flags & XFS_MOUNT_SMALL_INUMS)) {
-- 
2.34.1

